scripts/release/download_ERA5_TEC.py

```diff
@@ -0,0 +1,331 @@
+    slc_files : Sequence[Path]
+        list of paths to the slc files
+    grib_dir : Path
+        Path to the grib directory
+    frame_id : int
+        DISP-S1 frame id
+
+    Returns
+    -------
+    tuple[list[Path], Bbox]
+        The list of weather model files to be downloaded
+        and the bounding box of the frame
+    """
+    slc_file_list = group_by_date(slc_files)
+    first_date = next(iter(slc_file_list))
+    acquisition_time = get_zero_doppler_time(slc_file_list[first_date][0])
+    hour = closest_weather_model_hour(acquisition_time)
+
+    date_list = [
+        datetime.datetime.strftime(key[0], "%Y%m%d") for key in slc_file_list.keys()
+    ]
+
+    epsg, bounds = get_frame_bbox(frame_id)
+
+    if epsg != 4326:
+        bounds = transform_bounds(CRS.from_epsg(epsg), CRS.from_epsg(4326), *bounds)
+
+    snwe = (bounds[1], bounds[3], bounds[0], bounds[2])
+    area = snwe2str(snwe)
+
+    # grib file list
+    grib_files = []
+    for d in date_list:
+        if area:
+            grib_file = grib_dir / f"ERA5{area}_{d}_{hour}.grb"
+        else:
+            grib_file = grib_dir / f"ERA5_{d}_{hour}.grb"
+        grib_files.append(grib_file)
+
+    return grib_files, snwe
+
+
+def _add_buffer(snwe: tuple[float, float, float, float]):
+    s, n, w, e = snwe
+
+    min_buffer = 1
+    # lat/lon0/1 --> SNWE
+    S = np.floor(min(s, n) - min_buffer).astype(int)
+    N = np.ceil(max(s, n) + min_buffer).astype(int)
+    W = np.floor(min(w, e) - min_buffer).astype(int)
+    E = np.ceil(max(w, e) + min_buffer).astype(int)
+    return S, N, W, E
+
+
+def snwe2str(snwe: tuple[float, float, float, float]):
+    """Get area extent in string."""
+    if not snwe:
+        return None
+
+    S, N, W, E = _add_buffer(snwe)
+
+    area = ""
+    area += f"_S{abs(S)}" if S < 0 else f"_N{abs(S)}"
+    area += f"_S{abs(N)}" if N < 0 else f"_N{abs(N)}"
+    area += f"_W{abs(W)}" if W < 0 else f"_E{abs(W)}"
+    area += f"_W{abs(E)}" if E < 0 else f"_E{abs(E)}"
+
+    return area
+
+
+def closest_weather_model_hour(sar_acquisition_time: datetime.datetime) -> str:
+    """Find closest available time of weather product from SAR acquisition time.
+
+    Parameters
+    ----------
+    sar_acquisition_time: datetime.datetime
+        SAR data acquisition time
+
+    Returns
+    -------
+    grib_hr: str
+        time of closest available weather product in hour
+    """
+    # get hour/min of SAR acquisition time
+    # sar_time = datetime.strptime(sar_acquisition_time, "%Y-%m-%d %H:%M:%S.%f").hour
+    sar_time = sar_acquisition_time.hour
+
+    # find closest time in available weather products
+    grib_hr_list = np.arange(0, 24)
+    grib_hr = int(min(grib_hr_list, key=lambda x: abs(x - sar_time)))
+
+    # add zero padding
+    return f"{grib_hr:02d}"
+
+
+def main(iargs=None):
+    """Download ERA5 weather model files and TEC maps for all the given SLCs."""
+    parser = _get_cli_args()
+    args = parser.parse_args(args=iargs)
```

Comment:
do we need the `iargs` for integrating with something else? Or is it just leftover from mintpy?

Reply:
no we don't need it, I'll remove it